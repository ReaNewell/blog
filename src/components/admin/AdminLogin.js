import React from 'react';

class AdminLogin extends React.Component {
    render() {
        return (
            <div>Please login, sir.</div>
        )
    }
}

export default AdminLogin;