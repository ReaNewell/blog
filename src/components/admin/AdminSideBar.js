import React from 'react';
import { Link } from 'react-router-dom';

export class AdminSideBar extends React.Component {
    chooseActiveButton = (buttonPath) => {
        return this.props.path == buttonPath ? "admin-sidebar__button--active" : "admin-sidebar__button";
    }
    render() {
        return (
            <div className="admin-sidebar">
                <h3 className="admin-sidebar__welcome">Hello, Rea</h3>
                <Link className={this.chooseActiveButton('create')} to="/admin">Create Post</Link>
                <Link className={this.chooseActiveButton('edit')} to="/admin/edit-posts">Edit Posts</Link>
                <Link className={this.chooseActiveButton('settings')} to="/admin/settings">Settings</Link>
                <button className="admin-sidebar__button--logout">Logout</button>
            </div>
        )
    }
}

export default AdminSideBar;