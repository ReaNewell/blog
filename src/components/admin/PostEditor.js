import React from 'react';
import { connect } from 'react-redux';
import { startAddPost } from '../../actions/posts';
import { init } from 'pell';
import 'pell/dist/pell.css';

class PostEditor extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            error: "",
            title: "",
            html: null
        }
    };
    componentDidMount = () => {
        this.editor = init({
            element: document.getElementById('editor'),
            onChange: html => this.setState({ html }),
            actions: ['bold', 'underline', 'italic', 'heading1', 'heading2']
        })
    }
    onTitleChange = (e) => {
        const title = e.target.value;
        this.setState(() => ({ title }));
    }
    submitPost = () => {
        const image = document.getElementsByClassName('post-editor__image-input')[0].files[0];

        if (!this.state.html || !this.state.title) {
            this.setState(() => ({ error: "You must have a title and body."}));
        } else {
            this.setState(() => ({ error: ""}));
            this.props.startAddPost({
                title: this.state.title,
                body: this.state.html
            }, image);
            this.setState(() => ({ title: ""}));
            this.setState(() => ({ html: ""}));
        }
    };
    render() {
        return (
            <div className="post-editor">
                <input className="post-editor__title-input" onChange={this.onTitleChange} placeholder="Article Title" type="text"/>
                <div id="editor" className="pell" />
                <input 
                    accept='.jpg, .jpeg' 
                    className="post-editor__image-input" 
                    type='file'/>
                <button className="post-editor__button" onClick={this.submitPost}>Submit Post</button>
            </div>
        )
    };
}
const mapDispatchToProps = (dispatch) => ({
    startAddPost: (postInfo, image) => dispatch(startAddPost(postInfo, image))
});

export default connect(undefined, mapDispatchToProps)(PostEditor);