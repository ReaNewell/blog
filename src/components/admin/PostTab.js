import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { startRemovePost } from '../../actions/posts';

class PostTab extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            warning: false
        }
    }
    closeWarning = () => {
        this.setState(() => ({ warning: false }));
    }
    showWarning = () => {
        this.setState(() => ({ warning: true }));
    }
    removePost = (id, image) => {
        this.props.startRemovePost(id, image);
    }
    render() {
        return (
            <div>
                <h3>{this.props.post.title}</h3>
                <p>{this.props.post.postDate.month}/{this.props.post.postDate.day}/{this.props.post.postDate.year}</p>
                <Link to={`/admin/edit-posts/${this.props.post.title}`}>Edit Post</Link>
                {this.state.warning ? (
                    <div>
                        <p>Are you sure?</p>
                        <button onClick={this.removePost(this.props.post.id, this.props.post.postPictureName)}>
                            Yes
                        </button>
                        <button onClick={this.closeWarning}>No</button>
                    </div>
                ) : (
                    <button onClick={this.showWarning}>Delete Post</button>
                )}
            </div>
        )
    }
}
const mapDispatchToProps = (dispatch) => ({
    startRemovePost: (id, image) => dispatch(startRemovePost(id, image))
})

export default connect(undefined, mapDispatchToProps)(PostTab);