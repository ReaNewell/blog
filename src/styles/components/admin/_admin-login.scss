.admin-login {
    align-items: center;
    display: flex;
    flex-direction: column;
    justify-content: center;
    height: 100vh;
    width: 100vw;

    &__title {
        font-family: 'Chivo', sans-serif;
        font-size: $l-size;
        font-style: italic;
        font-weight: 700;
        margin: 1.5rem;
    }
}

.login-modal {
    background: linear-gradient(70deg, #67DFE7 20%, #4ACCFF 80%);
    display: grid;
    flex-direction: column;
    grid-template-areas:    '.' 
                            'email' 
                            'email-input' 
                            'password' 
                            'password-input' 
                            '.' 
                            'button' 
                            '.';
    min-height: 40rem;
    min-width: 40rem;

    &__label {
        align-self: center;
        color: $white;
        font-family: 'Chivo', sans-serif;
        font-size: $l-size;
        font-style: italic;
        justify-self: center;

        &:first-of-type {
            grid-area: email;
        }
        &:not(:first-of-type) {
            grid-area: password;
        }
    }

    &__input {
        align-self: center;
        background: rgba(255, 255, 255, 0.5);
        border: none;
        font-family: 'Chivo', sans-serif;
        font-style: italic;
        justify-self: center;
        height: 3rem;
        padding-left: 1rem;
        width: 75%;

        &:focus {
            outline: double white;
        }

        &#email-input {
            grid-area: email-input;
        }
        &#password-input {
            grid-area: password-input; 
        }
    }

    &__button {
        align-self: center;
        background: #00b6FF;
        border: none;
        color: $white;
        font-family: 'Chivo', sans-serif;
        font-size: $l-size;
        justify-self: center;
        grid-area: button;
        width: 30%;

        &:focus {
            outline: double white;
        }
    }
}

@media (max-width: $tablet-breakpoint) {
    .admin-login {
        background: linear-gradient(70deg, #67DFE7 20%, #4ACCFF 80%);
    }

    .login-modal {
        background: none;
        box-sizing: border-box;
        min-width: 90vw;

        &__button {
            width: 40%;
        }
    }
}