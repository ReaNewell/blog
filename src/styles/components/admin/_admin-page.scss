.dashboard {
    display: grid;
    grid-template-columns: 20rem auto;
    min-height: 100vh;
    width: 100vw;
}

.admin-page {
    padding: $l-size;

    .admin-page__title {
        margin: 0;
    }
    .admin-page__subtitle {
        margin: $l-size 0 $m-size 0;
    }
}