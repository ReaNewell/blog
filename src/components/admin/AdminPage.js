import React from 'react';
import AdminSideBar from './AdminSideBar';

class AdminPage extends React.Component {
    render() {
        return (
            <div>
                <AdminSideBar />
                <div>
                    <h1>Admin Page</h1>
                </div>
            </div>
        )
    }
}

export default AdminPage;