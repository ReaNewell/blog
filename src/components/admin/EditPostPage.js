import React from 'react';
import { connect } from 'react-redux';
import AdminSideBar from './AdminSideBar';
import PostEditor from './PostEditor';
import PostTab from './PostTab';

export class EditPostPage extends React.Component {
    render() {
        return (
            <div className="dashboard">
                <AdminSideBar path="edit"/>
                { this.props.currentPost ? (
                    <div>
                        <h1>Edit Posts</h1>
                        <h2>Make changes to article, then save.</h2>
                        <PostEditor 
                            history = {this.props.history} 
                            post={this.props.currentPost}
                            updating={true}
                        />
                    </div>
                ) : (
                    <div className="admin-page">
                        <h1 className="admin-page__title">Edit Posts</h1>
                        <h2 className="admin-page__subtitle">Posted Articles</h2>
                        <div>
                            { this.props.posts.map( post => (
                                <PostTab 
                                    key={post.id}
                                    post={post}
                                />
                            ))}
                        </div>
                    </div>
                )}
            </div>
        )
    }
}
const mapStateToProps = (state, props) => ({
    currentPost: props.match.params.title ? state.posts.find((post) => post.title === props.match.params.title.replace('-', ' ')) : null,
    posts: state.posts
})

export default connect(mapStateToProps)(EditPostPage);