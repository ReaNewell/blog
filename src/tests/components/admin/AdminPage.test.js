import React from 'react';
import { shallow } from 'enzyme';
import { AdminPage } from '../../../components/admin/AdminPage';

test('should render AdminPage component correctly', () => {
  const wrapper = shallow(<AdminPage />);
  expect(wrapper).toMatchSnapshot();
});