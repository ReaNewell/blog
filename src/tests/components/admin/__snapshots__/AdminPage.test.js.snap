// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AdminPage component correctly 1`] = `
<div
  className="dashboard"
>
  <AdminSideBar
    path="create"
  />
  <div
    className="admin-page"
  >
    <h1
      className="admin-page__title"
    >
      Admin Page
    </h1>
    <h2
      className="admin-page__subtitle"
    >
      Create A Post
    </h2>
    <Connect(PostEditor) />
  </div>
</div>
`;
